### PR TITLE
KH2: Update all instances of multiworld.option_name to option.option_name

### PR DESCRIPTION
**Pull request by JaredWeakStrike**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
changes all instances of the deprecated multiworld.option_name[player] to options.option_name

## How was this tested?
ran the tests and genned seeds and was not getting the deprecation error

### CODE REVIEW

File: worlds/kh2/Rules.py [hunk 1]
```diff
@@ -270,34 +270,34 @@ def set_kh2_rules(self) -> None:
...
+            final_xemnas_location.access_rule = lambda state: state.has(ItemName.Bounty, self.player, self.world.options.BountyRequired.value) and\
+                                  state.has(ItemName.LuckyEmblem, self.player, self.world.options.LuckyEmblemsRequired.value)
+            if self.world.options.FinalXemnas:
                 self.multiworld.completion_condition[self.player] = lambda state: state.has(ItemName.Victory, self.player, 1)
             else:
-                self.multiworld.completion_condition[self.player] = lambda state: state.has(ItemName.Bounty, self.player, self.multiworld.BountyRequired[self.player].value) and \
-                                                                                  state.has(ItemName.LuckyEmblem, self.player, self.multiworld.LuckyEmblemsRequired[self.player].value)
+                self.multiworld.completion_condition[self.player] = lambda state: state.has(ItemName.Bounty, self.player, self.world.options.BountyRequired.value) and \
+                                                                                  state.has(ItemName.LuckyEmblem, self.player,  self.world.options.LuckyEmblemsRequired.value)
```
Comment (Exempt-Medic) [inline worlds/kh2/Rules.py:300]: There's an extra space here
```suggestion
                                                                                  state.has(ItemName.LuckyEmblem, self.player, self.world.options.LuckyEmblemsRequired.value)
```

**Comment by Exempt-Medic** [comment]
Besides the one space mentioned in my previous comment, this looks fine to me, though I don't really feel qualified to put an approval on it. Did 200 generations and compared some spoiler logs before and after. No errors occurred, the logs matched, and KH2 was no longer listed in the list of worlds assigning options through `option_definitions`.

**Comment by JaredWeakStrike** [comment]
> PR has merge conflicts @JaredWeakStrike

No more merge conflicts

**Comment by JaredWeakStrike** [comment]
dont know how I requested a review. Sorry about that

**Review by qwint — APPROVED** [review]
Visual code review looks good, most changes are very straightforward moving to the new api. Also did checkout and run tests on my computer and confirm that no Options warnings exist for KH2 in the output.

did not do any generation/playtests as i don't know or own the game.

**Review by Silvris — APPROVED** [review]
LGTM. Option uses primary use `.value` so unlikely for anything to break. Did not test functionally.

**Review by black-sliver — APPROVED** [review]
Thanks!

Did not test either, but I'm hopeful that the author and CI would have found mistakes.

**Pull request merged**